client/java/src/main/java/io/openlineage/client/transports/HttpTransport.java

```diff
@@ -202,9 +208,7 @@ public HttpTransport build() {
       final HttpConfig httpConfig = new HttpConfig();
       httpConfig.setUrl(uri);
       httpConfig.setAuth(tokenProvider);
-      if (httpClient != null) {
```

Comment:
Not sure why you removed this - if we're getting passed another, externally configured HttpClient we should use it.

Reply:
Also this fails tests, since we pass mock client for tests this way 😉

Reply:
yeah, that was not intentional :s